src/components/views/elements/AppTile.js

```diff
@@ -100,20 +130,46 @@ export default React.createClass({
         }, "mx_IntegrationsManager");
     },
 
+    /* If user has permission to modify widgets, delete the widget, otherwise revoke access for the widget to load in the user's browser
+    */
     _onDeleteClick: function() {
-        console.log("Delete widget %s", this.props.id);
-        this.setState({deleting: true});
-        MatrixClientPeg.get().sendStateEvent(
-            this.props.room.roomId,
-            'im.vector.modular.widgets',
-            {}, // empty content
-            this.props.id,
-        ).then(() => {
-            console.log('Deleted widget');
-        }, (e) => {
-            console.error('Failed to delete widget', e);
-            this.setState({deleting: false});
-        });
+        if (this._canUserModify()) {
+            console.log("Delete widget %s", this.props.id);
+            this.setState({deleting: true});
+            MatrixClientPeg.get().sendStateEvent(
+                this.props.room.roomId,
+                'im.vector.modular.widgets',
+                {}, // empty content
```

Comment:
Nope, we changed the widget API a while back to be 1 state event per widget. https://docs.google.com/document/d/1TiWNDcEOULeRYQpkJHQDjgIW32ohIJSi5MKv9oRdzCo/edit